### PR TITLE
Adding the foundation for uniswap nfts

### PR DESCRIPTION
**Pull request by IanWoodard**
Laying the groundwork for our integrating with Uniswap NFTs by allowing users to deposit them as collateral. A few things to note:
- We fetch the uniswap nft positions on the borrow page since the fetching requires us to obtain all of the uniswap nft positions at once. This is fine due to our use of multicall meaning we only have to make a few calls to fetch many uniswap nft positions. Moreover, it enables us to filter the list of uniswap nft positions for the selected pool. 
- While I considered using the UniswapPosition type, it did not the granularity needed to filter by pair without wrapping it in more context. The good news about the new UniswapNFTPosition type is that it can easily be converted to a UniswapPosition; however, the UniswapPosition needs more context to be converted into a UniswapNFTPosition.
- I located and subsequently fixed a bug that was the result of passing the Token type through the multicall context, which stripped it of its instance methods. The fix involved passing the addresses and rebuilding the tokens from the other side.

### CODE REVIEW

File: earn/src/data/Token.ts
Comment (haydenshively) [inline earn/src/data/Token.ts:28]: We do not need to check `decimals`. Checking `chainId` and `address` guarantees equality.

File: earn/src/data/Uniswap.ts
Comment (haydenshively) [inline earn/src/data/Uniswap.ts:32]: My personal convention is to just call these `lower` and `upper`

**Review by haydenshively — DISMISSED** [review]
LGTM. Fetches look clean, nice work on that. Couple nits but don't wait for another review before merging

File: earn/src/data/Uniswap.ts
Comment (IanWoodard) [reply earn/src/data/Uniswap.ts:32]: I was following the naming convention from the contract.

**Pull request merged**
